build.zig

```diff
@@ -0,0 +1,423 @@
+                try nvim_headers.append(b.fmt("{s}{s}", .{ s, entry.name }));
+                if (api_export and !std.mem.eql(u8, "ui_events.in.h", entry.name)) {
+                    try api_headers.append(b.path(b.fmt("src/nvim/{s}{s}", .{ s, entry.name })));
+                }
+            }
+        }
+    }
+
+    const gen_config = b.addWriteFiles();
+
+    const version_lua = gen_config.add("nvim_version.lua", lua_version_info(b));
+
+    var config_str = b.fmt("zig build -Doptimize={s}", .{@tagName(optimize)});
+    if (cross_compiling) {
+        config_str = b.fmt("{s} -Dcross -Dtarget={s} (host: {s})", .{ config_str, try t.linuxTriple(b.allocator), try b.graph.host.result.linuxTriple(b.allocator) });
+    }
+
+    const versiondef_step = b.addConfigHeader(.{ .style = .{ .cmake = b.path("src/versiondef.h.in") } }, .{
+        .NVIM_VERSION_MAJOR = version.major,
+        .NVIM_VERSION_MINOR = version.minor,
+        .NVIM_VERSION_PATCH = version.patch,
+        .NVIM_VERSION_PRERELEASE = version.prerelease,
+        .VERSION_STRING = "TODO", // TODO(bfredl): not sure what to put here. summary already in "config_str"
+        .CONFIG = config_str,
+    });
+    _ = gen_config.addCopyFile(versiondef_step.getOutput(), "auto/versiondef.h"); // run_preprocessor() workaronnd
+
+    const isLinux = tag == .linux;
+    const modernUnix = tag.isDarwin() or tag.isBSD() or isLinux;
+
+    const ptrwidth = t.ptrBitWidth() / 8;
+    const sysconfig_step = b.addConfigHeader(.{ .style = .{ .cmake = b.path("cmake.config/config.h.in") } }, .{
+        .SIZEOF_INT = t.cTypeByteSize(.int),
+        .SIZEOF_INTMAX_T = t.cTypeByteSize(.longlong), // TODO
+        .SIZEOF_LONG = t.cTypeByteSize(.long),
+        .SIZEOF_SIZE_T = ptrwidth,
+        .SIZEOF_VOID_PTR = ptrwidth,
+
+        .PROJECT_NAME = "nvim",
+
+        .HAVE__NSGETENVIRON = tag.isDarwin(),
+        .HAVE_FD_CLOEXEC = modernUnix,
+        .HAVE_FSEEKO = modernUnix,
+        .HAVE_LANGINFO_H = modernUnix,
+        .HAVE_NL_LANGINFO_CODESET = modernUnix,
+        .HAVE_NL_MSG_CAT_CNTR = isLinux,
+        .HAVE_PWD_FUNCS = modernUnix,
+        .HAVE_READLINK = modernUnix,
+        .HAVE_STRNLEN = modernUnix,
+        .HAVE_STRCASECMP = modernUnix,
+        .HAVE_STRINGS_H = modernUnix,
+        .HAVE_STRNCASECMP = modernUnix,
+        .HAVE_STRPTIME = modernUnix,
+        .HAVE_XATTR = isLinux,
+        .HAVE_SYS_SDT_H = false,
+        .HAVE_SYS_UTSNAME_H = modernUnix,
+        .HAVE_SYS_WAIT_H = false, // unused
+        .HAVE_TERMIOS_H = modernUnix,
+        .HAVE_WORKING_LIBINTL = isLinux,
+        .UNIX = modernUnix,
+        .CASE_INSENSITIVE_FILENAME = tag.isDarwin() or tag == .windows,
+        .HAVE_SYS_UIO_H = modernUnix,
+        .HAVE_READV = modernUnix,
+        .HAVE_DIRFD_AND_FLOCK = modernUnix,
+        .HAVE_FORKPTY = modernUnix and !tag.isDarwin(), // also on Darwin but we lack the headers :(
+        .HAVE_BE64TOH = modernUnix and !tag.isDarwin(),
```

Comment:
If these are merely missing headers, but the function implementations are in libSystem, then please file a Zig issue so we can include the headers in the distribution.